src/pools/D3MAaveDaiPool.sol

```diff
@@ -0,0 +1,203 @@
+    function getReserveData(address asset) external view returns (
+        uint256,    // configuration
+        uint128,    // the liquidity index. Expressed in ray
+        uint128,    // variable borrow index. Expressed in ray
+        uint128,    // the current supply rate. Expressed in ray
+        uint128,    // the current variable borrow rate. Expressed in ray
+        uint128,    // the current stable borrow rate. Expressed in ray
+        uint40,     // last updated timestamp
+        address,    // address of the adai interest bearing token
+        address,    // address of the stable debt token
+        address,    // address of the variable debt token
+        address,    // address of the interest rate strategy
+        uint8       // the id of the reserve
+    );
+}
+
+interface RewardsClaimerLike {
+    function claimRewards(address[] calldata assets, uint256 amount, address to) external returns (uint256);
+}
+
+contract D3MAaveDaiPool is ID3MPool {
+
+    uint256 constant RAY  = 10 ** 27;
+
+    LendingPoolLike          public immutable pool;
+    RewardsClaimerLike       public immutable rewardsClaimer;
+    ATokenLike               public immutable stableDebt;
+    ATokenLike               public immutable variableDebt;
+    ATokenLike               public immutable adai;
+    TokenLike                public immutable asset; // Dai
+    address                  public           king;  // Who gets the rewards
+
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MAaveDaiPool/not-authorized");
+        _;
+    }
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+    event Collect(address indexed king, address[] assets, uint256 amt);
+
+    constructor(address hub_, address dai_, address pool_, address _rewardsClaimer) public {
+        pool = LendingPoolLike(pool_);
+        asset = TokenLike(dai_);
+
+        // Fetch the reserve data from Aave
+        (,,,,,,, address adai_, address stableDebt_, address variableDebt_, ,) = LendingPoolLike(pool_).getReserveData(dai_);
+        require(adai_ != address(0), "D3MAaveDaiPool/invalid-adai");
+        require(stableDebt_ != address(0), "D3MAaveDaiPool/invalid-stableDebt");
+        require(variableDebt_ != address(0), "D3MAaveDaiPool/invalid-variableDebt");
+
+        adai = ATokenLike(adai_);
+        stableDebt = ATokenLike(stableDebt_);
+        variableDebt = ATokenLike(variableDebt_);
+        rewardsClaimer = RewardsClaimerLike(_rewardsClaimer);
+
+        ATokenLike(adai_).approve(pool_, type(uint256).max);
```

Comment:
The aDai approval appears to be unnecessary because the pool has permission to burn aTokens.

Reply:
I think this comment is old, but just in case, this is our pool contract, not Aave pool.

Reply:
done: 862e431

@gbalabasquer I added this in the constructor thinking i needed to give the aave pool to pull adai tokens and burn them from our pool. Turns out, aave pool can burn without my permission, so this is unnecessary